src/App.js

```diff
@@ -1,19 +1,295 @@
+  const [sortBy,setSortBy] = useState(searchParams.get('sortBy') || RELEASE_DATE)
+  const query = searchParams.get('query') || null;
+  const [searchString, setSearchString] = useState(query);
+
+
+  const [movies, setMovies] = useState([]);
+  const [genreList, setGenreList] = useState([]);
+  
+  const [selectedMovieData,setSelectedMovieData] = useState(null);
+  const [showMovieInfoPanel,setShowMovieInfoPanel] = useState(false);
+  
+  
+  const [deleteMovieDialog, setDeleteMovieDialog] = useState({
+    isOpen: false,
+    movieInfo: null
+  });
+  const [movieFormResult, setMovieFormResult] = useState({
+    msg: null,
+    isOpen: false,
+    isSuccess: true
+  })
+
+  const openDeleteMovieDialog = (movieInfo) => {
+    setDeleteMovieDialog({
+      isOpen: true,
+      movieInfo: movieInfo
+    });
+  };
+
+  const closeDeleteMovieDialog = () => {
+    setDeleteMovieDialog({
+      isOpen: false,
+      movieInfo: null
+    });
+  };
+
+  const openMovieFormResultDialog = (msg, isSuccess) => {
+    setMovieFormResult({
+      isSuccess: isSuccess,
+      msg: msg,
+      isOpen: true
+    })
+  };
+
+  const closeMovieFormResultDialog = () => {
+    setMovieFormResult({
+      isSuccess: true,
+      msg: null,
+      isOpen: false
+    })
+  };
+  
+
+  useEffect(() => {
+    console.log("useEffect Setting Genre Menu's", selectedGenre, fetchCurrentQueryParams(searchParams))
+    let tempList = [];
+    fetchGenreList()
+    .then((response) => {
+      response.data.data.map(i=>{
+        tempList = tempList.concat(i.genres)
+        return i.genres;
+      });
+      tempList = Array.from(new Set(tempList));
+      setGenreList(tempList.toSorted())
+      return tempList;
+    })
+    .catch((err) => {
+      console.log('error', err)
+     });
+  }, [])
+
+  
+  useEffect(() => {
+    updateCurrentURL()
+    console.log("useEffect called.>>")
+    callMoviewAPI(selectedGenre, sortBy, query);
+  }, [selectedGenre, sortBy, query, searchString])
+
+  const updateCurrentURL=()=>{
+    const queryParams ={activeGenre: selectedGenre, sortBy: sortBy};
+    if(searchString){
+      queryParams['query'] = searchString;
+    }
+    setSearchParams(queryParams);
+  }
+
+  const callMoviewAPI = (selectedGenre, sortBy,  searchString) =>{
+    let searchQuerParam = new URLSearchParams();
+    searchQuerParam.set("offset", 0);
+    searchQuerParam.set("limit", 20);
+    if(selectedGenre && !searchString){
+      searchQuerParam.set("filter", selectedGenre);
+    }
+    searchQuerParam.set("sortBy", sortBy);
+    searchQuerParam.set("sortOrder", "desc");
+    if (searchString) {
+      searchQuerParam.set("searchBy", "title");
+      searchQuerParam.set("search", searchString);
+    }
+    axios.get(HOST+"/movies?"+searchQuerParam.toString())
+    .then((response) => {
+      setMovies(response.data.data);
+    })
+    .catch((err) => {
+      console.log('error', err)
+     });
```

Comment:
Better to move this to a service file and use async await. Always prefer using ES6 or above features